apps/studio/src/server/modules/whitelist/whitelist.service.ts

```diff
@@ -0,0 +1,66 @@
+import { TRPCError } from "@trpc/server"
+
+import { db } from "../database"
+
+export const isEmailWhitelisted = async (email: string) => {
+  const lowercaseEmail = email.toLowerCase()
+
+  // Step 1: Check if the exact email address is whitelisted
+  const exactMatch = await db
+    .selectFrom("Whitelist")
+    .where("email", "=", lowercaseEmail)
+    .where(({ eb }) =>
+      eb.or([eb("expiry", "is", null), eb("expiry", ">", new Date())]),
+    )
+    .select(["id"])
+    .executeTakeFirst()
+
+  if (exactMatch) {
+    return true
+  }
+
+  // Step 2: Check if the exact email domain is whitelisted
+  const emailParts = lowercaseEmail.split("@")
```

Comment:
do we need whitelisting by domain? I thought we decided to whitelist only by exact emails going forward

Reply:
This is for government agencies so they can self-service use Studio without having to send in a request for explicit whitelisting, otherwise the ops load will be quite high if we have to explicitly whitelist each individual emails.